localstripe/resources.py

```diff
@@ -1400,6 +1395,70 @@ def _api_list_all(cls, url, customer=None, limit=None):
         return li
 
 
+class InvoiceLineItem(StripeObject):
+    object = 'line_item'
+    _id_prefix = 'il_'
+
+    def __init__(self, item):
+        try:
+            assert isinstance(item, (InvoiceItem, SubscriptionItem))
+        except AssertionError:
+            raise UserError(400, 'Bad request')
+
+        # All exceptions must be raised before this point.
+        super().__init__()
+
+        self.type = \
+            'invoiceitem' if isinstance(item, InvoiceItem) else 'subscription'
+
+        if self.type == 'subscription':
+            self.subscription_item = item.id
+            self.subscription = item._subscription
+        elif self.type == 'invoiceitem':
+            self.invoice_item = item.id
+            self.subscription = None
```

Comment:
Maybe I suggest the following? It seems more readable to me:
```python
        if self.type == 'subscription':
            self.subscription_item = item.id
            self.subscription = item._subscription
            self.plan = item.plan
            self.quantity = item.quantity
            self.amount = item._calculate_amount()
            self.currency = item.plan.currency
            ...
            subscription_obj = Subscription._api_retrieve(item._subscription)
            self.period = dict(start=subscription_obj.current_period_start,
                               end=subscription_obj.current_period_end)
        elif self.type == 'invoiceitem':
            self.invoice_item = item.id
            self.subscription = None
            self.plan = None
            self.quantity = 1
            self.currency = item.currency
            ...
            self.amount = item.amount
            self.period = item.period
```

Reply:
Indeed it is. Done!